src/components.js

```diff
@@ -155,20 +157,21 @@ export class Component extends React.Component {
   }
 
   getValue() {
-    return this.getTransformer().parse(this.state.value)
+    const value = this.state ? this.state.value : this.getTransformer().format(this.props.value)
+    return this.getTransformer().parse(value)
   }
 
   isValueNully() {
     return Nil.is(this.getValue())
   }
 
   removeErrors() {
-    this.setState({ hasError: false })
+    this.setState({ hasError: false, error: '' })
   }
 
   validate() {
     const result = t.validate(this.getValue(), this.props.type, this.getValidationOptions())
-    this.setState({ hasError: !result.isValid() })
+    this.setState({ hasError: !result.isValid(), error: this.getError() })
```

Comment:
This could be a problem. e.g. if there is an error hasError will become true but in the moment when this.getError() is called hasError is still true. The result is that component will have error style but there is no error message. Possible solution is to wait until state has changed and than set state for error or, pass to getError param true to ignore the hasError state.

Reply:
That's right, thanks.

@gcanti 
any opinions regarding this PR?